parseInt.js

```diff
@@ -1,24 +1,28 @@
 /**
- * Run a small snippet of JavaScript during a mabl flow/journey
+ * Parse text and extract integer values from that text. 
  * @param {object} - mablInputs Object containing input
  *                   variables (mablInputs.variables.user)
  * @param {function} callback - The callback function
  */
 function mablJavaScriptStep(mablInputs, callback) {
 
-    let element = document.querySelector('REPLACE_WITH_CSS_SELECTOR');
+  // grab the element
+  let element = document.querySelector('REPLACE_WITH_CSS_SELECTOR');
 
-    if (element) {
-        let elementText = element.innerText;
-        let parsed = parseInt(elementText);
-        
-        if (isNaN(parsed)) {
-         throw Error('Inner text does not contain an integer');
-        } else {
-          callback(parsed);
-        }
-     
-    } else {
-        throw Error('Element cannot be found');
-    }
-  }
\ No newline at end of file
+  // check to see if the element exists
+  if (!element) {
+    throw Error('Element cannot be found');
+  }
+
+  // grab the inner text and parse that text for integers
+  let elementText = element.innerText;
+  let parsed = parseInt(elementText, 10);
+
+  // check to see if the inner text contains integers, by checking the return value of the parsed text
+  if (isNaN(parsed)) {
+    throw Error('Inner text does not contain an integer');
```

Comment:
given the behavior of `parseInt()` this is really `Inner text does not start with an integer.` (e.g., `'25abcd'` will give you `25` but `'abcd25'` will be `NaN`)

Reply:
misread the documentation there, might change the entirety of the code to parse each char of the string and extract the integers rather than using parseInt().. Thank you @johnkbrew

Reply:
no problem!

you can probably do that pretty easily with a regex (i.e., match on `/\d+/`) and then use `parseInt` on that (or on the first match, if any; probably doesn't make sense to deal with multiple integer matches in the string for a general snippet).

Reply:
So should this text change to "starts with" now?